### ISSUE TITLE
All requests apart from get account return empty array.

### ISSUE THREAD

**Issue by Jameshudson** [domdomegg/halifax-share-dealing-sdk#5]
I'm running the following code. Which from my understanding of the README file should return the default account and the dividends history.

```
require('dotenv').config();

const halifaxShareDealingSdk = require('halifax-share-dealing-sdk');

const sd = halifaxShareDealingSdk({
    site: process.env.SITE,
    USERNAME: process.env.USERNAME,
    PASSWORD: process.env.PASSWORD,
    MEMORABLE_INFORMATION: {
      "Your mother's FIRST name": process.env.MOTHER_FIRST_NAME,
      "Your father's FIRST name": process.env.FATHER_FIRST_NAME,
      'Your place/town of birth': process.env.PLACE_TOWN_OF_BIRTH,
      'The name of your first school': process.env.FIRST_SCHOOL,
    }
});

sd.login()
    .then(sd.getDefaultAccount)
    .then(sd.getDividendHistory)
    .then(console.dir)
```

I get the following in the command output. I am running it in a node:10 docker container (If that makes any difference)
`
Got memorable information question and understood prompt
Successfully logged in
Got default account
Got dividend history
[]
`

**Comment by domdomegg** [comment]
Thanks for this report. I can reproduce this issue and have identified the root cause.

This is happening as Halifax must have changed something on their server rendering side which alters how tables work - they no longer use `tbody` elements. The current selectors expect `tbody` elements, so miss the necessary data. It'd be good to have contract tests to catch this earlier but unfortunately I don't think it's possible to get a sandbox account or similar :(

I'll try to fix the selectors and release a new version soon

Let me know if there's anything else I can help you with :)

**Comment by domdomegg** [comment]
I've dug a bit deeper in fixing it and `getDividendHistory` actually appears to work for me. `getStatements` is actually broken.

However, I am planning to change the behaviour to get all the dividends paid for the account rather than the fairly arbitrary 'whatever the page by default returns', which may help for your purposes. The documentation states this method 'Returns an array of historical dividends relating to the account', so I don't consider this a breaking change.

**Comment by domdomegg** [comment]
Fix released in [halifax-share-dealing@1.5.5](https://www.npmjs.com/package/halifax-share-dealing-sdk/v/1.5.5), available from npm now.

Going to close this, if there is anything else feel free to reopen this issue or create a new one. Thanks again for the report 👍

**Comment by Jameshudson** [comment]
Nice one, thanks for being so quick 👍
